tracing/src/main/java/com/palantir/remoting2/tracing/Tracers.java

```diff
@@ -96,11 +100,14 @@ public static Runnable wrap(Runnable delegate) {
         @Override
         public V call() throws Exception {
             Trace originalTrace = Tracer.copyTrace();
+            String originalMdcTraceIdValue = MDC.get(MDC_KEY);
             Tracer.setTrace(trace);
+            MDC.put(MDC_KEY, trace.getTraceId());
```

Comment:
ah, we could rely on thread-local storage like in Tracer. We'd have to do the same for the user/session/token id info too.